src/pyodide/internal/loadPackage.ts

```diff
@@ -17,11 +16,16 @@ import { parseTarInfo } from "pyodide-internal:tar";
 import { default as DiskCache } from "pyodide-internal:disk_cache";
 import { createTarFS } from "pyodide-internal:tarfs";
 
-async function decompressArrayBuffer(arrBuf) {
-  return await new Response(new Response(arrBuf).body.pipeThrough(new DecompressionStream("gzip"))).arrayBuffer();
+async function decompressArrayBuffer(arrBuf: ArrayBuffer): Promise<ArrayBuffer> {
+  const resp = new Response(arrBuf);
+  if (resp && resp.body) {
+    return await new Response(resp.body.pipeThrough(new DecompressionStream("gzip"))).arrayBuffer();
+  } else {
+    throw "Failed to decompress array buffer";
```

Comment:
Don't throw bare strings
```suggestion
    throw new Error("Failed to decompress array buffer");
```